tool/speed.cc

```diff
@@ -660,10 +660,75 @@ static bool SpeedKEM(const std::string &name, int nid, const std::string &select
   return true;
 }
 
-
 static bool SpeedKEM(std::string selected) {
   return SpeedKEM("Kyber512", NID_KYBER512, selected);
 }
+
+static bool SpeedDigestSignNID(const std::string &name, int nid,
+                            const std::string &selected) {
+  if (!selected.empty() && name.find(selected) == std::string::npos) {
+    return true;
+  }
+
+  // Setup CTX for Sign/Verify Operations
+  BM_NAMESPACE::UniquePtr<EVP_PKEY_CTX> pkey_ctx(EVP_PKEY_CTX_new_id(nid, nullptr));
+
+  // Setup CTX for Keygen Operations
+  if (!pkey_ctx || EVP_PKEY_keygen_init(pkey_ctx.get()) != 1) {
+    return false;
+  }
+
+  EVP_PKEY *key = NULL;
+
+  TimeResults results;
+  if (!TimeFunction(&results, [&pkey_ctx, &key]() -> bool {
+        return EVP_PKEY_keygen(pkey_ctx.get(), &key);
+      })) {
+    return false;
+  }
+  results.Print(name + " keygen");
+
+  // Setup CTX for Sign operations
+  bssl::ScopedEVP_MD_CTX md_ctx;
+
+  // message to be signed
+  static const uint8_t msg[32] = {0};
+  size_t msg_len = 32;
+
+  // to keep this function generic, we obtain the signature size (different for
+  // each algorithm) at run time by attempting a sign with a NULL signature.
+  // The sign algorithm must support calling NULL to obtain the signature length
+  size_t sig_len = 0;
+  EVP_DigestSignInit(md_ctx.get(), NULL, NULL, NULL, key);
+  EVP_DigestSign(md_ctx.get(), NULL, &sig_len, msg, msg_len);
+  std::unique_ptr<uint8_t[]> signature(new uint8_t[sig_len]);
+
+
+  if (!TimeFunction(&results, [&md_ctx, &key, &signature, &sig_len, msg_len ]() -> bool {
+        return EVP_DigestSignInit(md_ctx.get(), NULL, NULL, NULL, key) &&
```

Comment:
Do we need to call sign init each iteration?

Reply:
you are correct, removed in [f865b54](https://github.com/awslabs/aws-lc/pull/724/commits/f865b5454bc50474974894a87307ec1621b09a3c)